core/js/typespecs/app_types.ts

```diff
@@ -380,56 +387,51 @@ export const aclObjectTypeEnum = z
 export type AclObjectType = z.infer<typeof aclObjectTypeEnum>;
 
 const aclBaseSchema = generateBaseTableSchema("acl");
-export const aclObjectSchema = z.strictObject({
-  id: aclBaseSchema.shape.id,
-  object_type: aclObjectTypeEnum,
-  object_id: z
-    .string()
-    .uuid()
-    .describe("The id of the object the ACL applies to"),
-  restrict_object_type: z
-    .optional(z.union([aclObjectTypeEnum, z.null()]))
-    .describe(
-      "Optionally restricts the permission grant to just the specified object type",
-    ),
-  _object_org_id: z
-    .string()
-    .uuid()
-    .describe("The organization the ACL's referred object belongs to"),
-  created: aclBaseSchema.shape.created,
-});
-const aclUserObjectSchema = z.strictObject({
-  user_id: z.string().uuid().describe("Id of the user the ACL applies to"),
-});
-const aclGroupObjectSchema = z.strictObject({
-  group_id: z.string().uuid().describe("Id of the group the ACL applies to"),
-});
-const aclPermissionObjectSchema = z.strictObject({
-  permission: permissionEnum.describe("Permission the ACL grants"),
-});
-const aclRoleObjectSchema = z.strictObject({
-  role_id: z.string().uuid().describe("Id of the role the ACL grants"),
-});
-
 export const aclSchema = z
-  .union([
-    aclObjectSchema
-      .merge(aclUserObjectSchema)
-      .merge(aclPermissionObjectSchema)
-      .openapi("UserPermissionAcl"),
-    aclObjectSchema
-      .merge(aclUserObjectSchema)
-      .merge(aclRoleObjectSchema)
-      .openapi("UserRoleAcl"),
-    aclObjectSchema
-      .merge(aclGroupObjectSchema)
-      .merge(aclPermissionObjectSchema)
-      .openapi("GroupPermissionAcl"),
-    aclObjectSchema
-      .merge(aclGroupObjectSchema)
-      .merge(aclRoleObjectSchema)
-      .openapi("GroupRoleAcl"),
-  ])
+  .strictObject({
```

Comment:
The whole "disjoint union" thing was too complicated for the stainless SDK generator, because it wasn't able to find the "id" property needed for pagination by looking at each of the union members. I felt it might be too confusing to enumerate each possible combination of user/group/permission/role to be worth the type safety, so I just flattened the object.